src/aotcompile.cpp

```diff
@@ -453,11 +453,11 @@ void jl_dump_native_impl(void *native_code,
     TheTriple.setOS(llvm::Triple::MacOSX);
 #endif
     std::unique_ptr<TargetMachine> TM(
-        jl_TargetMachine->getTarget().createTargetMachine(
+        jl_ExecutionEngine->getTargetMachine().getTarget().createTargetMachine(
```

Comment:
this seems a little bit odd. Should we use a TargetMachineBuilder?

Reply:
We probably should, especially since we can choose to not expose the actual TargetMachine itself from the JIT engine.

Reply:
Update on this: my attempt at using a TargetMachineBuilder failed during building on macos, so I'm thinking it's better to revisit rewriting this particular sequence of instructions at a later date.